apps/src/code-studio/pd/teacher_application/teacher_application.jsx

```diff
@@ -0,0 +1,374 @@
+          >
+            {answer}
+          </Radio>
+        );
+      } else if (type === 'check') {
+        return (
+          <Checkbox
+            value={answer}
+            label={answer}
+            key={i}
+            name={groupName}
+          >
+            {answer}
+          </Checkbox>
+        );
+      }
+    });
+
+    if (allowOther) {
+      const otherDiv = (
+        <div style={{display: 'inline'}}>
+          <span style={{marginRight: '5px'}}>
+            Other:
+          </span>
+          <input type="text" id={groupName + 'Other'}/>
+        </div>
+      );
+
+      if (type === 'radio') {
+        options.push((
+          <Radio
+            key={answers.length + 1}
+            name={groupName}
+          >
+            {otherDiv}
+          </Radio>
+        ));
+      } else if (type === 'check') {
+        options.push((
+          <Checkbox
+            key={answers.length + 1}
+            name={groupName}
+          >
+            {otherDiv}
+          </Checkbox>
+        ));
+      }
+    }
+
+    return (
+      <div>
+        <ControlLabel>
+          {label}
+        </ControlLabel>
+        {options}
+      </div>
+    );
+  },
+
+  generateTeacherInformationSection() {
+    return (
+      <div>
+        {this.buttonList('check', 'Grades served at your school', 'gradesAtSchool', grades)}
+        <h2>
+          Section 2: Teacher Information
+        </h2>
+        <FieldGroup
+          id="firstName"
+          label="First name"
+          type="text"
+        />
+        <FieldGroup
+          id="preferredFirstName"
+          label="Preferred First Name"
+          type="text"
+        />
+        <FieldGroup
+          id="lastName"
+          label="Last Name"
+          type="text"
+        />
+        <FieldGroup
+          id="schoolEmail"
+          label="Your school email address"
+          type="email"
+        />
+        <FieldGroup
+          id="schoolEmail"
+          label="Your personal email address (we may need to contact you during the summer)"
+          type="email"
+        />
+        <FieldGroup
+          id="phoneNumber"
+          label="Preferred phone number"
+          type="text"
+        />
+        {this.buttonList('radio', 'Gender Identity', 'genderIdentity', ['Female', 'Male', 'Other', 'Prefer not to answer'])}
+        {this.buttonList('check', 'What grades are you teaching in the current 2016-17 school year? (select all that apply)', 'grades2016', grades)}
+        {this.buttonList('check', 'What subjects are you teaching in the current 2016-17 school year? (select all that apply)', 'subjects2016', subjects, true)}
+        {this.buttonList('check', 'What grades are you teaching in the current 2017-18 school year? (select all that apply)', 'grades2017', grades)}
+        {this.buttonList('check', 'What subjects are you teaching in the current 2016-17 school year? (select all that apply)', 'subjects2017', subjects, true)}
```

Comment:
This is a little cumbersome. What about making `buttonList` a functional stateless component like `FieldGroup`? Then you can do something like:

```jsx
<ButtonList
  type='radio'
  label='Gender Identity'
  groupName='genderIdentity'
  answers={['Female, 'Male', 'Other', 'Prefer not to answer']}
/>
```